tools/build-tool/code/tooldebian.pas

```diff
@@ -0,0 +1,192 @@
+
+  CreateDir(ShareDir + PathDelim + 'menu');
+  TextWriter := TTextWriter.Create(ShareDir + PathDelim + 'menu' + PathDelim + Manifest.Name);
+  TextWriter.Write(
+    '?package(' + Manifest.Name + '): \' + NL +
+    'needs="X11" \' + NL +
+    'section="' + Manifest.DebianSection + '" \' + NL +
+    'title="' + Manifest.Caption + '" \' + NL +
+    'command="bash -c ''cd ' + PathToExecutableUnix + ' && ./' + Manifest.ExecutableName + '''" \' + NL +
+    'icon="' + PathToIconFileUnix +'"'
+    );
+  FreeAndNil(TextWriter);
+
+  CreateDir(ShareDir + PathDelim + 'applications');
+  TextWriter := TTextWriter.Create(ShareDir + PathDelim + 'applications' + PathDelim + Manifest.ExecutableName + '.desktop');
+  TextWriter.Write(
+    '[Desktop Entry]' + NL +
+    'Version=' + Manifest.Version.DisplayValue + NL +
+    'Terminal=false' + NL +
+    'Exec=bash -c ''cd ' + PathToExecutableUnix + ' && ./' + Manifest.ExecutableName + '''' + NL +
+    'Icon=' + PathToIconFileUnix + NL +
+    'Type=Application' + NL +
+    'Categories=' + Manifest.DebianCategories + NL +
+    'Name=' + Manifest.Caption + NL +
+    'Comment=' + Manifest.DebianComment
+    );
+  FreeAndNil(TextWriter);
+
+  CreateDir(PackageFolder + PathDelim + 'DEBIAN');
+  TextWriter := TTextWriter.Create(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'control');
+  TextWriter.Write(
+    'Package: ' + Manifest.Name + NL +
+    'Version: ' + Manifest.Version.DisplayValue + NL +
+    'Section: ' + Manifest.DebianInstallFolder + NL +
+    'Priority: optional' + NL +
+    'Architecture: ' + CpuToArchitectureString(Cpu) + NL +
+    'Maintainer: ' + Manifest.Author + NL +
+    'Installed-Size: ' + IntToStr(BinariesSize div 1024) + NL +
+    'Depends: libopenal1, libpng16-16, zlib1g, libvorbis0a, libvorbisfile3, libfreetype6, libgl1-mesa-dri, libgtk2.0-0' + NL +
+    'Description: ' + Manifest.Caption + NL +
+    ' ' + Manifest.DebianComment + NL //final new line
+    );
+  FreeAndNil(TextWriter);
+
+  // Post-installation running - assign executable permissions
+
+  TextWriter := TTextWriter.Create(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'postinst');
+  TextWriter.Write(
+    '#!/bin/sh' + NL + NL +
+    'chmod +x ' + PathToExecutableUnix + '/' + Manifest.ExecutableName + NL + NL +
+    'exit 0'
+    );
+  FreeAndNil(TextWriter);
+  fpChmod(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'postinst', &555);
+
+  // Workaround, we need to execute a shell script somehow
+
+  TextWriter := TTextWriter.Create('castle-engine-output' + PathDelim + 'package-debian.sh');
+  TextWriter.Write(
+    'cd castle-engine-output' + NL +
+    'cd ' + PackageFileName + NL +
+    'find -type f | egrep -v ''^\./DEBIAN'' | xargs --replace=hh -n1 md5sum "hh" | sed ''s/\ \.\///'' > DEBIAN/md5sums' + NL +
+    'cd ..' + NL +
+    'dpkg-deb --build ' + PackageFileName
+    );
+  FreeAndNil(TextWriter);
+
+  RunCommand('/bin/bash', ['castle-engine-output/package-debian.sh'], OutString);
```

Comment:
It is rather unclean to create a shell script, only to execute it :) Moreover this script has a problem: it doesn't use `set -e` so it will work OK even if some command in the middle will fail (see http://redsymbol.net/articles/unofficial-bash-strict-mode/ for nice discussion on how to make bash scripts more reliable).

Better to use a series of `RunCommandSimple` calls (or other routines from `ToolCommonUtils` or FPC `Process`). If you need pipes for 1 line, run this line with `bash -c`.

Like this:

```
RunCommandSimple(PackageDir, 'bash', ['-c', 
  'find -type f | egrep -v ''^\./DEBIAN'' | xargs --replace=hh -n1 md5sum "hh" | sed ''s/\ \.\///'' > DEBIAN/md5sums']);

// you may need to fix quoting above. Since you don't create shell script, you have one less thing to escape.

RunCommandSimple(TempOutputPath(Project.Path), 'dpkg-deb', ['--build', PackageFileName]);
```

Reply:
Done in cc5a5a32d5ef9cf591182ced9edc7b128479f273